### ISSUE TITLE
VMSS update failure AttributeError: 'NoneType' object has no attribute 'pop' 

### ISSUE THREAD

**Issue by alex-goncharov** [Azure/azure-cli#12119]
## Describe the bug

**Command Name**
`az vmss update`

**Errors:**
```
'NoneType' object has no attribute 'pop'
Traceback (most recent call last):
python3.8/site-packages/knack/cli.py, ln 206, in invoke
    cmd_result = self.invocation.execute(args)
cli/core/commands/__init__.py, ln 608, in execute
    raise ex
cli/core/commands/__init__.py, ln 666, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
cli/core/commands/__init__.py, ln 659, in _run_job
    six.reraise(*sys.exc_info())
...
cli/core/commands/__init__.py, ln 306, in __call__
    return self.handler(*args, **kwargs)
cli/core/commands/arm.py, ln 546, in handler
    remove_properties(instance, arg_values)
cli/core/commands/arm.py, ln 921, in remove_properties
    list_to_remove_from.pop(int(list_index))
AttributeError: 'NoneType' object has no attribute 'pop'
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az vmss update --resource-group {} --name {} --remove {} {}`

## Environment Summary
```
macOS-10.15.3-x86_64-i386-64bit
Python 3.8.1
Shell: bash

azure-cli 2.0.81

Extensions:
azure-firewall 0.1.5
aks-preview 0.4.23

```

**Comment by arrownj** [comment]
@alex-goncharov , thanks for reporting, we are looking at this.

**Comment by SnoWolfT** [comment]
Open the ICM to PG. PG suggests to update the steps to reproduce the issue in this Github link. Follow this to fix the error handling issue in azure cli.

I make tests and find this issue can be reproduced as below. The value/list of loadBalancerInboundNatPools is empty. So when the command tries to “remove” the index 0 of the empty property, “loadBalancerInboundNatPools”, the error message will be reproduced.

az vmss update --resource-group MC_120021022001352_askazupdate_eastus --name aks-agentpool-15442772-vmss --remove virtualMachineProfile.networkProfile.networkInterfaceConfigurations[0].ipConfigurations[0].loadBalancerInboundNatPools 0
 

If we try to remove one property which has relevant index/sub-property, the command should be fine. For example, I use the property, loadBalancerBackendAddressPools.

PS Azure:\> $vmss=Get-AzVmss -ResourceGroupName "MC_120021022001352_askazupdate_eastus" -Name "aks-agentpool-15442772-vmss"
 
PS Azure:\> $vmss.virtualMachineProfile.networkProfile.networkInterfaceConfigurations[0].ipConfigurations[0].loadBalancerBackendAddressPools
 
Id
--
/subscriptions/7bd52e62-08f9-4c6c-a3ab-45489bc7bf6b/resourceGroups/MC_120021022001352_askazupdate_eastus/providers/Microsoft.Network/loadBalancers/kubernetes/backendAddressPools/aksOutboundBackendPool
/subscriptions/7bd52e62-08f9-4c6c-a3ab-45489bc7bf6b/resourceGroups/MC_120021022001352_askazupdate_eastus/providers/Microsoft.Network/loadBalancers/kubernetes/backendAddressPools/Kubernetes
 
PS Azure:\> $vmss.virtualMachineProfile.networkProfile.networkInterfaceConfigurations[0].ipConfigurations[0].loadBalancerBackendAddressPools[0]
 
Id
--
/subscriptions/7bd52e62-08f9-4c6c-a3ab-45489bc7bf6b/resourceGroups/MC_120021022001352_askazupdate_eastus/providers/Microsoft.Network/loadBalancers/kubernetes/backendAddressPools/aksOutboundBackendPool
 
Azure:/
PS Azure:\> $vmss.virtualMachineProfile.networkProfile.networkInterfaceConfigurations[0].ipConfigurations[0].loadBalancerBackendAddressPools[1]
 
Id
--
/subscriptions/7bd52e62-08f9-4c6c-a3ab-45489bc7bf6b/resourceGroups/MC_120021022001352_askazupdate_eastus/providers/Microsoft.Network/loadBalancers/kubernetes/backendAddressPools/kubernetes
 

Then I use the az cli to remove the index 0 of loadBalancerBackendAddressPools without error message.

tom@Azure:~$ az vmss update --resource-group MC_120021022001352_askazupdate_eastus --name aks-agentpool-15442772-vmss --remove virtualMachineProfile.networkProfile.networkInterfaceConfigurations[0].ipConfigurations[0].loadBalancerBackendAddressPools 0
{
  "additionalCapabilities": null,
  "automaticRepairsPolicy": null,
  "doNotRunExtensionsOnOverprovisionedVms": false,
…
 

As expected, I can see the only the former index 1 exists. Currently it becomes the index 0 since the former index 0 is removed.

PS Azure:\> $vmss.virtualMachineProfile.networkProfile.networkInterfaceConfigurations[0].ipConfigurations[0].loadBalancerBackendAddressPools
 
Id
--
/subscriptions/7bd52e62-08f9-4c6c-a3ab-45489bc7bf6b/resourceGroups/MC_120021022001352_askazupdate_eastus/providers/Microsoft.Network/loadBalancers/kubernetes/backendAddressPools/kubernetes
 

If I try to remove one index doesn’t exist but the property isn’t empty, I will encounter the corresponding error message.

tom@Azure:~$ az vmss update --resource-group MC_120021022001352_askazupdate_eastus --name aks-agentpool-15442772-vmss --remove virtualMachineProfile.networkProfile.networkInterfaceConfigurations[0].ipConfigurations[0].loadBalancerBackendAddressPools 9
index 9 doesn't exist on loadBalancerBackendAddressPools
 

But if I remove all the index in loadBalancerBackendAddressPools and run the command again, I will encounter the error handling issue.

tom@Azure:~$ az vmss update --resource-group MC_120021022001352_askazupdate_eastus --name aks-agentpool-15442772-vmss --remove virtualMachineProfile.networkProfile.networkInterfaceConfigurations[0].ipConfigurations[0].loadBalancerBackendAddressPools 0
The command failed with an unexpected error. Here is the traceback:
 
'NoneType' object has no attribute 'pop'
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 608, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 666, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 659, in _run_job

**Comment by arrownj** [comment]
Hi @alex-goncharov @SnoWolfT , this issue is fixed in version 2.4.0. You can upgrade your version and have a try. Thanks
